core/secret.go

```diff
@@ -40,9 +39,7 @@ func NewSecretFromHostEnv(name string) (*Secret, error) {
 type SecretID string
 
 func NewSecretID(name, plaintext string) (SecretID, error) {
-	digestBytes := sha256.Sum256([]byte(plaintext))
```

Comment:
Is this change intentional? we're not using `plaintext` anymore in this function?

Reply:
It was a way to get the cache busted when the plaintext of a secret gets changed, but it leaked some info (the secret has changed), and it made the support in Dockerfile painful to users (they had to use some templated Dockerfile, with the secret ID already encoded with the digest of the plaintext).
Here it's more compatible with simple secret ID for buildkit. The secret name becomes the secret ID. `setSecret` will just register the secret name/ID to its plaintext in the `secret.Store` buildkit session attachable.

Reply:
got it, thx! since we're not using `plaintext` anymore in that function, could we remove it? or that breaks some user facing API's?